Core/src/main/java/com/paypal/android/core/api/EligibilityAPI.kt

```diff
@@ -0,0 +1,44 @@
+package com.paypal.android.core.api
+
+import android.util.Log
+import com.paypal.android.core.CoreConfig
+import com.paypal.android.core.api.models.APIResult
+import com.paypal.android.core.api.models.Eligibility
+import com.paypal.android.core.graphql.common.GraphQlClient
+import com.paypal.android.core.graphql.fundingEligibility.FundingEligibilityQuery
+import com.paypal.android.core.graphql.fundingEligibility.models.FundingEligibilityIntent
+import com.paypal.android.core.graphql.fundingEligibility.models.SupportedCountryCurrencyType
+import com.paypal.android.core.graphql.fundingEligibility.models.SupportedPaymentMethodsType
+
+class EligibilityAPI(
+    private val graphQlClient: GraphQlClient,
+    private val coreConfig: CoreConfig
+) {
+
+    suspend fun checkEligibility(): APIResult<Eligibility> {
+        val fundingEligibilityQuery = FundingEligibilityQuery(
+            clientId = coreConfig.clientId,
+            fundingEligibilityIntent = FundingEligibilityIntent.CAPTURE,
+            currencyCode = SupportedCountryCurrencyType.USD,
+            enableFunding = listOf(SupportedPaymentMethodsType.VENMO)
+        )
+            val response = graphQlClient.executeQuery(fundingEligibilityQuery)
+            Log.d(TAG, response.toString())
+            return if (response.data != null) APIResult.Success(
+                Eligibility(
+                    isCreditCardEligible = response.data.fundingEligibility.card.eligible,
+                    isPayLaterEligible = response.data.fundingEligibility.payLater.eligible,
+                    isPaypalCreditEligible = response.data.fundingEligibility.credit.eligible,
+                    isPaypalEligible = response.data.fundingEligibility.paypal.eligible,
+                    isVenmoEligible = response.data.fundingEligibility.venmo.eligible,
+                )
+            )
+            else APIResult.Failure(
+                message = "Failure in fetching data"
+            )
```

Comment:
kinda like I mentioned before, if the calling function is `suspend` it stand to reason that failures are thrown via `exceptions`.
Conceptually, `checkElegibility()` should return just `Eligibility` or throw a `PayPalSDKError` exception.

@sarahkoop @sshropshire thoughts?

Reply:
I agree if we are returning success or throwing everywhere else in the SDK, we should be consistent here.

Reply:
@jcnoriega same +1.

Reply:
the documentation is more of a guide to follow, and david wrote it in `swift` just to show the idea of what we need to do, its not meant to be a technical guide, as each platform has its own kirks (like `listener` in Android vs `delegate` in iOS). So, keep in mind with Android, `suspend` either throws a result (`Eligbility`) or an exception. Everything under `APIFailure` should be wrapped in a `PayPalSDKError`.

Reply:
@jcnoriega didn't get what you're trying to say,
yep code is in iOS that's enum alternative in kotlin is sealded class,
you're saying return `APIFailure< PayPalSDKError>` instead of `APIFailure` having it's own things?
and suspend doesn't need to throw exception, in that case we're asking other end to make all handling of exceptions
we can huddle to avoid back & forth